### DISCUSSION TITLE
Dispatch events in JS components

### DISCUSSION THREAD

**Discussion by CleitonDeLima** [themesberg/flowbite#286] [Ideas]
Are there development plans to dispatch events in JS components?
This is very useful to integrate with other js libraries, such as [htmx](https://htmx.org/) or [_hyperscript](https://hyperscript.org/).
Example:

```js
const modal = new Modal(targetEl, options);

modal.show();  // dispatch CustomEvent('modal.show')

modal.addEventListener('modal.show', function () {
  alert('modal open');
})
```

```html
<div hx-get="/foo/" hx-trigger="modal.show" >
  ...
</div>
```

**Comment by zoltanszogyenyi** [comment]
Hello @CleitonDeLima,

This will be seriously considered for the v2.0 update.

You're also more than welcome to contribute to this.
